src/main/scala/edu/ie3/mobsim/model/ChargingBehavior.scala

```diff
@@ -74,39 +75,51 @@ object ChargingBehavior extends LazyLogging {
         logger.debug(
           s"${ev.getId} arrives at charging hub and wants to start charging."
         )
-        ev.getDestinationPoi.nearestChargingStations.keys.headOption
-          .map(_.getUuid)
+        (
+          ev.destinationPoi.nearestChargingStations.keys.headOption
+            .map(_.uuid),
+          None
+        )
       } else {
         /* Update charging prices memory of EV to have a reference for the prices of specific charging stations */
-        ev.getDestinationPoi.nearestChargingStations.keys.foreach { cs =>
-          currentPricesAtChargingStations
-            .get(cs.getUuid)
-            .map(ev.updateChargingPricesMemory)
-        }
+
+        val prices = ev.destinationPoi.nearestChargingStations.keys
+          .map { cs =>
+            currentPricesAtChargingStations(cs.uuid)
+          }
+          .to(Queue)
+
+        val evWithUpdatedPriceMemory = ev.updateChargingPricesMemory(prices)
```

Comment:
`evWithUpdatedPriceMemory` is not used with `doesEvWantToCharge` and `createRatingsForChargingStations`, which I believe can lead to wrong results.

Please fix this and also create a test case that fails now but succeeds once this is fixed, as this is very important to function the same way as before.